Bottom: 5c5cfcad27124334c1abd2e2d4c4ed62d642ab44
Top:    0ce3ac0fb486fcfbaba8663ba4c8b1e26a74c229
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-24 17:16:30 +0100

Refresh of hid-gtk-track-previous-state-o

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7230aa3..2a66baf 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -407,6 +407,7 @@ ghid_show_crosshair (gboolean show)
 {
   gint x, y;
   static gint x_prev = -1, y_prev = -1;
+  static gboolean hca, hca_prev = FALSE;
   static GdkGC *xor_gc;
   static GdkColor cross_color;
 
@@ -429,7 +430,7 @@ ghid_show_crosshair (gboolean show)
   if (x_prev >= 0)
     {
       draw_crosshair (xor_gc, x_prev, y_prev);
-      if (ghidgui->auto_pan_on && have_crosshair_attachments ())
+      if (ghidgui->auto_pan_on && hca_prev)
 	{
 	  gdk_draw_rectangle (gport->drawing_area->window, xor_gc, TRUE,
 			      0, y_prev - VCD, VCD, VCW);
@@ -445,7 +446,8 @@ ghid_show_crosshair (gboolean show)
   if (x >= 0 && show)
     {
       draw_crosshair (xor_gc, x, y);
-      if (ghidgui->auto_pan_on && have_crosshair_attachments ())
+      hca = have_crosshair_attachments ();
+      if (ghidgui->auto_pan_on && hca)
 	{
 	  gdk_draw_rectangle (gport->drawing_area->window, xor_gc, TRUE,
 			      0, y - VCD, VCD, VCW);
@@ -458,6 +460,7 @@ ghid_show_crosshair (gboolean show)
 	}
       x_prev = x;
       y_prev = y;
+      hca_prev = hca;
     }
   else
     x_prev = y_prev = -1;
